lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -1442,22 +1443,30 @@ def install_packages_on(hosts, package_hash, options = {})
           return true
         end
 
-        def ruby_command(host)
-          "env PATH=\"#{host['privatebindir']}:${PATH}\" ruby"
-        end
-
-        def gem_command(host, type = 'aio')
+        def get_command(command_name, host, type = 'aio')
           if type == 'aio'
             if host['platform'] =~ /windows/
-              "env PATH=\"#{host['privatebindir']}:${PATH}\" cmd /c gem"
+              "env PATH=\"#{host['privatebindir']}:${PATH}\" cmd /c #{command_name}"
             else
-              "env PATH=\"#{host['privatebindir']}:${PATH}\" gem"
+              "env PATH=\"#{host['privatebindir']}:${PATH}\" #{command_name}"
             end
           else
-            on(host, 'which gem').stdout.chomp
+            on(host, "which #{command_name}").stdout.chomp
           end
         end
 
+        def ruby_command(host, type = 'aio')
+          get_command('ruby', host, type)
```

Comment:
This will cause us to execute `cmd /c ruby` on windows which I guess works. Normally we only need to wrap `cmd` for puppet batch files due to the way that `environment.bat` sets up the `PATH`.

Reply:
We were doing this for bundler in some cases, and gem, so I figured it wouldn't be a bad idea to add it for consistency with these different commands? Do you think it'd be better to change it back?